app/editor/src/features/admin/sources/DataSource.tsx

```diff
@@ -66,9 +70,28 @@ export const DataSource: React.FC<IDataSourceProps> = (props) => {
             }
           >
             <Outlet />
-            <Col alignContent="flex-start" alignItems="flex-end">
+            <Row justify="flex-end">
               <Button type="submit">Save</Button>
-            </Col>
+              <Button onClick={toggle} variant={ButtonVariant.danger}>
+                Remove Data Source
+              </Button>
+            </Row>
+            <Modal
+              headerText="Confirm Removal"
+              body="Are you sure you wish to remove this data source?"
+              isShowing={isShowing}
+              hide={toggle}
+              type="delete"
+              confirmText="Yes, Remove It"
+              onConfirm={async () => {
+                try {
+                  await api.deleteDataSource(source);
```

Comment:
The data sources are in redux memory, this won't delete the item from the in-memory collection.

Reply:
Just to verify, does this line not cover that?
`store.storeDataSources(getSources().filter((ds) => ds.id !== result.id));`

Reply:
found within the `deleteDataSource` method

Reply:
It probably does!